src/csharp/Generator.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public class Generator : IDisposable
+    {
+        private IntPtr _generatorHandle;
+
+        public Generator(Model model, GeneratorParams generatorParams)
+        {
+            Result.VerifySuccess(NativeMethods.OgaCreateGenerator(model.Handle, generatorParams.Handle, out _generatorHandle));
+        }
+
+        public bool IsDone()
+        {
+            return NativeMethods.OgaGenerator_IsDone(_generatorHandle);
+        }
+
+        public void ComputeLogits()
+        {
+            Result.VerifySuccess(NativeMethods.OgaGenerator_ComputeLogits(_generatorHandle));
+        }
+
+        public void GenerateNextTokenTop()
+        {
+            Result.VerifySuccess(NativeMethods.OgaGenerator_GenerateNextToken_Top(_generatorHandle));
+        }
+
+        public int[] GetSequence(int index)
```

Comment:
Same comment, return span directly to native data, this reduces the amount of garbage and eliminates data copying.
Also implement this[index] that returns a span instead of the method. C# developers expect appropriate patterns.
Add an out of bounds check for the index argument.